lib/src/common/repos_service.dart

```diff
@@ -208,86 +247,192 @@ class RepositoriesService extends Service {
+    return PaginationHelper(_github)
+        .objects<Map<String, dynamic>, CommitComment>(
+      "GET",
+      "repos/${slug.fullName}/comments",
+      (i) => CommitComment.fromJSON(i),
+      statusCode: StatusCodes.OK,
+    );
+  }
+
+  /// Create a comment for a commit using its sha.
+  /// * [body]: The contents of the comment.
+  /// * [path]: Relative path of the file to comment on.
+  /// * [position]: Line index in the diff to comment on.
+  /// * [line]: **Deprecated**. Use position parameter instead. Line number in the file to comment on.
+  ///
+  /// https://developer.github.com/v3/repos/comments/#create-a-commit-comment
+  Future<CommitComment> createCommitComment(
+    RepositorySlug slug,
+    RepositoryCommit commit, {
+    @required String body,
+    String path,
+    int position,
+    @Deprecated('Use position parameter instead') int line,
+  }) async {
+    final Map<String, dynamic> data = createNonNullMap({
+      'body': body,
+      'path': path,
+      'position': position,
+      'line': line,
     });
+    return _github.postJSON<Map<String, dynamic>, CommitComment>(
+      "/repos/${slug.fullName}/commits/${commit.sha}/comments",
+      body: jsonEncode(data),
+      statusCode: 201,
+      convert: (i) => CommitComment.fromJSON(i),
+    );
   }
 
-  // TODO: Implement listComments: https://developer.github.com/v3/repos/comments/#list-commit-comments-for-a-repository
-  // TODO: Implement listCommitComments: https://developer.github.com/v3/repos/comments/#list-comments-for-a-single-commit
-  // TODO: Implement createComment: https://developer.github.com/v3/repos/comments/#create-a-commit-comment
-  // TODO: Implement getComment: https://developer.github.com/v3/repos/comments/#get-a-single-commit-comment
-  // TODO: Implement updateComment: https://developer.github.com/v3/repos/comments/#update-a-commit-comment
-  // TODO: Implement deleteComment: https://developer.github.com/v3/repos/comments/#delete-a-commit-comment
+  /// Retrieve a commit comment by its id.
+  ///
+  /// https://developer.github.com/v3/repos/comments/#get-a-single-commit-comment
+  Future<CommitComment> getCommitComment(RepositorySlug slug,
+      {@required int id}) async {
+    return _github.getJSON<Map<String, dynamic>, CommitComment>(
+      "/repos/${slug.fullName}/comments/id",
+      statusCode: 200,
+      convert: (i) => CommitComment.fromJSON(i),
+    );
+  }
+
+  /// Update a commit comment
+  /// * [id]: id of the comment to update.
+  /// * [body]: new body of the comment.
+  ///
+  /// Returns the updated commit comment.
+  ///
+  /// https://developer.github.com/v3/repos/comments/#update-a-commit-comment
+  Future<CommitComment> updateCommitComment(RepositorySlug slug,
+      {@required int id, @required String body}) async {
+    assert(body != null);
+    return _github.postJSON<Map<String, dynamic>, CommitComment>(
+      "/repos/${slug.fullName}/comments/${id.toString()}",
```

Comment:
#nit the string interpolation will automatically toString the id. This could just be
```suggestion
      "/repos/${slug.fullName}/comments/$id",
```